includes/class-amp-story-post-type.php

```diff
@@ -340,6 +344,83 @@ public static function remove_classic_editor_link( $actions, $post ) {
 		return $actions;
 	}
 
+	/**
+	 * Filters allowed CSS attributes to include positioning, display, etc.
+	 *
+	 * @param array $attr Array of allowed CSS attributes.
+	 *
+	 * @return array Filtered array of allowed CSS attributes.
+	 */
+	public static function filter_safe_style_css( $attr ) {
+
+		// Only continue if it's a REST request for amp_story post type.
+		if ( ! isset( $GLOBALS['wp']->query_vars['rest_route'] ) ) {
+			return $attr;
+		}
+
+		$route  = $GLOBALS['wp']->query_vars['rest_route'];
+		$search = '\/' . self::POST_TYPE_SLUG . '\/';
+		if ( ! preg_match( "/{$search}/i", $route ) ) {
+			return $attr;
+		}
+
+		$style_to_add = array(
+			'display',
+			'left',
+			'position',
+			'top',
+		);
+
+		foreach ( $style_to_add as $style ) {
+			$attr[] = $style;
+		}
+		return $attr;
+	}
+
+	/**
+	 * Filter user caps to add more permissions for Author being able to use AMP Stories properly.
+	 *
+	 * This is not ideal and should be removed once https://core.trac.wordpress.org/ticket/37134 has been resolved.
+	 *
+	 * @param array $allcaps All caps.
+	 * @param array $caps    Requested caps.
+	 * @param array $args    Cap args.
+	 * @return array All caps.
+	 */
+	public static function filter_user_has_cap_for_enabling_author_editing( $allcaps, $caps, $args ) {
+		if ( ! isset( $args[0] ) || 'unfiltered_html' !== $args[0] ) {
+			return $allcaps;
+		}
+
+		if ( ! current_user_can( 'edit_posts' ) ) {
+			return $allcaps;
+		}
+
+		// Let's get the post's ID from the request and see if it's an AMP Story.
+		if ( ! isset( $_REQUEST['rest_route'] ) ) { // phpcs:ignore WordPress.CSRF.NonceVerification.NoNonceVerification
```

Comment:
@swissspidy Here the `GLOBALS` is not always available when needed. Not sure what's a good approach here.

Reply:
`$_REQUEST['rest_route']` doesn't work when using pretty permalinks. So I guess we need to parse the request URI ourselves, because this runs before WP does its init...